File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    void inorder(TreeNode *root, vector<int> &vec){
        if(root!=NULL){
            inorder(root->left, vec);
            vec.push_back(root->val);
            inorder(root->right, vec);
        }
    }
    int minDiffInBST(TreeNode* root) {
      
        int small = INT_MAX;
        
        if(root==NULL) return 0;
        
        vector<int> vec;
        
        inorder(root, vec);
        
        for(int i=1; i<vec.size(); i++){
            
            int diff = abs(vec[i-1] - vec[i]);
            
            if(small > diff) small = diff;
            
        }
        
        return small;
        
        
    }
};

File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDiffInBST(TreeNode root) {
       List<Integer> list = new ArrayList<>();   
        inorder(root, list);
        int min = Integer.MAX_VALUE;
        for (int i = 1; i < list.size(); i++) {
            int dif = Math.abs(list.get(i - 1) - list.get(i));
            if (min > dif) 
                min = dif;
        }
        return min;   
    }
      static public void inorder(TreeNode root, List<Integer> list) {
      
        if(root==null)
            return;
        inorder(root.left, list);
                list.add(root.val);
        inorder(root.right, list);
    }
}